active-rfcs/0000-router-view-keep-alive-transitions.md

```diff
@@ -0,0 +1,101 @@
+- Start Date: 2020-04-20
+- Target Major Version: Vue Router 4.x
+- Reference Issues:
+- Implementation PR:
+
+# Summary
+
+Given the changes to functional components in Vue 3, `KeepAlive` and `Transition` usage combined with `RouterView` is no longer possible by simply wrapping `RouterView` with `KeepAlive`/`Transition`. Instead we need a way to directly provide the component rendered by `RouterView` to those components:
+
+```vue
+<router-view v-slot="{ Component, attrs }">
+  <transition :name="transitionName" mode="out-in">
+    <component v-if="Component" :is="Component" v-bind="attrs"></component>
+  </transition>
+</router-view>
+```
+
+# Motivation
+
+As described at https://github.com/vuejs/vue-next/issues/906#issuecomment-611080663, we need a new API to allow the usage of `KeepAlive` and other components accepting slots with `RouterView`. In order to implement such behavior we need to be able to **directly** wrap the component rendered by `RouterView`. The only way to do so by having access to the component rendered by `RouterView` and the _props_ passed to it.
+
+# Detailed design
+
+We can achieve this level of control through a slot:
+
+```vue
+<router-view v-slot="{ Component, attrs }">
+  <component :is="Component" v-bind="attrs"></component>
+</router-view>
+```
+
+In the example, `Component` is the _component definition_ that can be passed to the function `h` or to the `is` prop of `component`. `attrs` is a set of event listeners and props that `RouterView` **requires** in order to work properly. Namely, `onVnodeMounted`/`onVnodeUnmounted` and a `ref`. It also contains any _attribute_/_prop_ passed to `RouterView`. Let's say we pass a `class` to it, we will find it in `attrs`:
+
+```vue
+<router-view class="view" v-slot="{ attrs }">
```

Comment:
I don't see the point of this. It makes sense when using the non-slot syntax, but with a slot the user should simply add the class directly in the slot.

Reply:
It was more of a convenience thing, especially if there are `v-if` statements so the user can pass attributes directly in one single place but if we remove `attrs` then this is no longer necessary